### ISSUE TITLE
"New file creation" out of sync with calendar date

### ISSUE THREAD

**Issue by tebe-nigrelli** [mendhak/gpslogger#1058]
Steps to reproduce: in the "Logging details" menu select a "New file creation" option out of daily, monthly, and custom. When dates are saved, the file of TODAY may include dates from 11 pm of YESTERDAY or go up to 11pm of TODAY, despite logging running continuously.

**Comment by mendhak** [comment]
Yes I think there's definitely a bug here, I have myself seen a 'daily' file go up to 3 AM even 5 AM of the next day, before switching over. I'd guess there's some kind of file name being kept in session without checking if it's still valid. It'll take a bit of testing to get to it. I'll mark this as a bug for now to investigate soon.

**Comment by j-schneid** [comment]
Might this also have to do with the time zone? I have the impression, switching to a new file always happens at midnight UTC. I would find it more useful to be local time (including daylight savings if applicable).

**Comment by mendhak** [comment]
So I finally got around to some testing and I think this is likely behaving as expected. The time in the GPX file is UTC. That means it's possible to have log lines from yesterday/today in today/yesterday's file, depending on the time zone you are in. 

There is an option "Log time with timezone offset" which should use the local device's timezone instead of UTC, in the GPX file. Although the option exists I think it's technically an invalid GPX file.
